server/src/main/java/org/opensearch/search/startree/StarTreeFilter.java

```diff
@@ -164,7 +172,8 @@ private static StarTreeResult traverseStarTree(StarTreeValues starTreeValues, Ma
 
             String childDimension = dimensionNames.get(dimensionId + 1);
             StarTreeNode starNode = null;
-            if (globalRemainingPredicateColumns == null || !globalRemainingPredicateColumns.contains(childDimension)) {
+            if (globalRemainingPredicateColumns == null
```

Comment:
if((globalRemainingPredicateColumns == null
                || !globalRemainingPredicateColumns.contains(childDimension)) && !remainingGroupByColumns.contains(childDimension))

Predicate conditions should be grouped together